src/test/java/org/llorllale/cactoos/matchers/MatchesBeforeTest.java

```diff
@@ -70,24 +68,36 @@ public void mismatchesFromMatcher() {
         ).affirm();
     }
 
-    // @todo #133:30min Currently MatchesBefore directly throws a
-    //  TimeoutException but instead it should populate the mismatch
-    //  message so that it can be properly tested with Mismatches.
-    //  Unignore and update this test to validate it is correctly implemented
-    @Ignore
     @Test
     public void mismatchesFromTimeout() {
         final String val = "c";
         new Assertion<>(
             "must fail because of timeout",
             new MatchesBefore<>(10, new TextIs(val)),
             new Mismatches<>(
-                (Text) () -> {
+                () -> {
                     Thread.sleep(1000);
                     return val;
                 },
                 "Text with value \"c\" runs in less than <10L> milliseconds",
-                "Tto be determined"
+                "Timeout after <10L> milliseconds"
+            )
+        ).affirm();
+    }
+
+    @Test
+    public void mismatchesWhenExceptionThrown() {
+        final String val = "c";
+        new Assertion<>(
+            "must fail because of inner exception",
```

Comment:
@andreoss can you improve this message? I think it so vague.

Reply:
@fabriciofx Done